### ISSUE TITLE
Setup fiddle project in the WebdriverIO GitHub org

### ISSUE THREAD

**Issue by christian-bromann** [webdriverio/webdriverio#4503]
I know that [@klamping](https://github.com/klamping/trywebdriverio) as well as @WillBrock have been working on a web app that allows to build fiddle code with WebdriverIO and run it somewhere. In order to offer an offical fiddle app we need to have the code within the WebdriverIO GitHub org. Therefor we need to figure out:

1. @klamping would you mind to move your project into the WebdriverIO org?
2. @WillBrock How far did you implement your fiddle application?
3. With which one do we want to move forward?

**Comment by klamping** [comment]
For reference, my project code is here: https://github.com/klamping/trywebdriverio

I'm thinking I would rather start the fiddle project from scratch, as what I put together was very much a proof-of-concept and I haven't done any work on it since January. The project is still using wdio 4.x as well, so that would need to be updated. I will share my overall architecture though:

**Front-end:** 
* [Mustache based templates/views](https://github.com/klamping/trywebdriverio/blob/master/views/index.mustache)
* [Ace code editor](https://github.com/ajaxorg/ace)
* [Static images/css/js](https://github.com/klamping/trywebdriverio/tree/master/public)
* [CSS is custom built, no frameworks/libraries for that](https://github.com/klamping/trywebdriverio/blob/master/public/app.css)
* Terminal output looks to be just plain text inside of a `<pre>`. I think i wanted to use [xterm](https://github.com/xtermjs/xterm.js) at some point, but never got it going. 

**Back-end:**
* [ExpressJS serves the front-end code](https://github.com/klamping/trywebdriverio/blob/master/index.js)
* Communication between front-end/back-end is done via Web Sockets. Pretty happy with how it worked, although there were some difficulties along the way with timeouts and such. Overall I'd recommend sticking with it unless someone knows better :)
1. On page open, [WS connection is made to Express Server](https://github.com/klamping/trywebdriverio/blob/master/public/app.js#L94)
2. When user clicks "Run test", [it sends the test data/info to express](https://github.com/klamping/trywebdriverio/blob/master/public/app.js#L138)
3. Which in turn [handles the request](https://github.com/klamping/trywebdriverio/blob/master/index.js#L122):
4. [the `runTest` function](https://github.com/klamping/trywebdriverio/blob/master/lib/startRun.js) spawns a new docker process which starts the test run (it also manages messages back and forth from that process):
5. The docker process uses two images:
A. A custom wdio runner image built from [this folder](https://github.com/klamping/trywebdriverio/tree/master/docker/wdio) and [this dockerfile](https://github.com/klamping/trywebdriverio/blob/master/docker/Dockerfile)
B. The `selenium/standalone-chrome` from https://github.com/SeleniumHQ/docker-selenium. Could probably be switched to just a chromedriver image for quicker tests
6. As the test runs, the output is sent [from the docker images to express](https://github.com/klamping/trywebdriverio/blob/master/index.js#L38), which sends them back to the front-end via websockets

There are a couple other small features here and there but that's the main setup of it. I would love to tie this in with a technology like serverless, especially from a security aspect. Always worried me having this run on a Digital Ocean machine, even though it's theoretically all behind Docker. 

Attached is some feedback i got a while back about the website that I never found time to implement. Added for reference:
[Test Observations_TryLearnWebdriverIO.docx](https://github.com/webdriverio/webdriverio/files/3623117/Test.Observations_TryLearnWebdriverIO.docx)

Also, @TuHuynhVan took a stab at running a custom version of this for his needs and was able to make some progress. Haven't talked with him about it since March though.

**Comment by baruchvlz** [comment]
> I think i wanted to use xterm at some point, but never got it going.

Isn't this what https://repl.it uses? It's quite powerful and it can be more intriguing for the users than just plain text.

> Could probably be switched to just a chromedriver image for quicker tests

This [image](https://hub.docker.com/r/blueimp/chromedriver/dockerfile) seems pretty solid.

Maybe running it headless as well?
```js
capabilities: [{
    maxInstances: 5,

    browserName: 'chrome'
    'goog:chromeOptions': { // not 100% sure on the name
        args: ['--headless']
    }
}],
```

>  I would love to tie this in with a technology like serverless, especially from a security aspect

Since you said you wanted to re-work the project from scratch, you could use React with NextJS and use the [serverless option](https://nextjs.org/blog/next-8#serverless-nextjs). This will allow you to not have to worry about serving the front end code and also write React because React is good and handlebars is not (personal opinion).

**Comment by christian-bromann** [comment]
I was even thinking for the first iteration to run everything in the browser using Sauce Labs. We have user profiles that can run up to 30 sessions in parallel which should be enough for the beginning. Second iteration could be to allow to run things headless. This way we can ignore the "backend" part to get something useful running quickly.

**Comment by christian-bromann** [comment]
Closing in favor of #4709 - we should first find out what we want to build and then set up the project.
